### ISSUE TITLE
Add file bundling to elf to uf2 conversion

### ISSUE THREAD

**Issue by mattnite** [ZigEmbeddedGroup/uf2#1]
Implement [file containers](https://github.com/microsoft/uf2#file-containers) for uf2. There's already a build option for the user to select whether the source files should be bundled, which ends up hitting a panic if it's set to true right now. If you parse DWARF info, you'll be able to assemble a list of all used files, and then add them to the uf2 file using the `addFile()` function.

**Comment by pqcfox** [comment]
Would love to help contribute this feature! Is this available to work on?

**Comment by mattnite** [comment]
@foxmoder Yes it is! [here](https://github.com/ZigEmbeddedGroup/uf2/blob/c523a4d23469282f95658a879f5ba925757dc9d9/src/uf2.zig#L169) is the relevant code path. Let me know if you need any other guidance!
